frontend/src/api/tus.js

```diff
@@ -0,0 +1,50 @@
+import * as tus from "tus-js-client";
+import { tusEndpoint } from "@/utils/constants";
+import store from "@/store";
+import { removePrefix } from "./utils";
+
+// Make following configurable by envs?
+export const chunkSize = 20 * 1000 * 1000;
+const parallelUploads = 3;
+const retryDelays = [0, 3000, 5000, 10000, 20000];
+
+export async function upload(url, content = "", overwrite = false, onupload) {
+  return new Promise((resolve, reject) => {
+    var upload = new tus.Upload(content, {
+      endpoint: tusEndpoint,
+      chunkSize: chunkSize,
+      retryDelays: retryDelays,
+      parallelUploads: parallelUploads,
+      metadata: {
+        filename: content.name,
+        filetype: content.type,
+        overwrite: overwrite.toString(),
+        // url is URI encoded and needs to be for metadata first
+        destination: decodeURIComponent(removePrefix(url)),
+      },
+      headers: {
+        "X-Auth": store.state.jwt,
+      },
+      onError: function (error) {
+        reject("Upload failed: " + error);
+      },
+      onProgress: function (bytesUploaded) {
+        // Emulate ProgressEvent.loaded which is used by calling functions
+        // loaded is specified in bytes (https://developer.mozilla.org/en-US/docs/Web/API/ProgressEvent/loaded)
+        if (typeof onupload === "function") {
+          onupload({ loaded: bytesUploaded });
+        }
+      },
+      onSuccess: function () {
+        resolve();
+      },
+    });
+
+    upload.findPreviousUploads().then(function (previousUploads) {
+      if (previousUploads.length) {
+        upload.resumeFromPreviousUpload(previousUploads[0]);
```

Comment:
Will this only retry the very first upload. Since previousUploads is an array I would expect to have to iterate and retry them all. It looks like the intention is to prompt the user to select which upload they want to resume so we might need to add a popup to let the user manually resume uploads. Usage guide on this can be found [here](https://github.com/tus/tus-js-client/blob/master/docs/usage.md#example-let-user-select-upload-to-resume)

Reply:
FindPreviousUploads retrieves a list of uploads that have been previously started for this file. According to the documentation, a fingerprint of the file is used for findPreviousUploads. The fingerprint doesn't include the file's hash (see [here](https://github.com/tus/tus-js-client/blob/aed3e8420e3105610bebdb6f6de3322495079484/lib/browser/fileSignature.js#L27)) 
So, which information would you like to show the user before letting them make a decision on which upload to resume? Why not simply select the most recent upload or the largest one?